packages/scripts/utils/config.js

```diff
@@ -183,19 +183,29 @@ function getWebpackEntryPoints() {
 		return JSON.parse( process.env.WP_ENTRY );
 	}
 
-	// Continue only if the `src` directory exists.
-	if ( ! hasProjectFile( 'src' ) ) {
+	// Continue only if the source directory exists.
+	if ( ! hasProjectFile( process.env.WP_SRC_DIRECTORY ) ) {
+		log(
```

Comment:
It made me think that we could use a dynamic entry as documented in https://webpack.js.org/configuration/entry-context/#dynamic-entry. We would never run this code when the project overrides entry points.

> If a function is passed, then it will be invoked on every [make](https://webpack.js.org/api/compiler-hooks/#make) event.
> 
> Note that the making event triggers when webpack starts and for every invalidation when [watching for file changes](https://webpack.js.org/configuration/watch/).


In fact, this would solve a more pressing issue we recently discovered with @schmitzoide when working on a plugin with multiple blocks. Currently, the logic for finding entry points runs only once in the watch mode. Therefore you need to restart the development build whenever new blocks get introduced or scripts for the block change. Using a callback function should resolve the issue.

Reply:
Related comment I mentioned: https://github.com/WordPress/gutenberg/issues/38739#issuecomment-1039318204.

Reply:
@gziolo @ryanwelcher Hope this PR resolves the confusing behaviour for https://github.com/WordPress/gutenberg/issues/38739#issuecomment-1039318204

Reply:
I added commit 37ec581 to see if that resolves the issue.

Reply:
I can confirm that the message isn't shown when we use the default handling for entry points.

I can also confirm that entry points get dynamically updated in the watch mode when `block.json` changes.